aten/src/ATen/native/Distributions.cpp

```diff
@@ -234,8 +234,43 @@ Tensor& normal_out_cpu(Tensor& output, double mean, const Tensor& std, Generator
 }
 
 Tensor& normal_out_cpu(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  bool expandable = are_expandable(mean.sizes(), std.sizes());
+  bool empty_output = output.numel() == 0;
+
+  if (expandable) {
```

Comment:
A lot of this logic is duplicated among normal_out_cpu and normal_out_cuda. Could we write a helper function to deduplicate the logic?

Something like
```
bool is_legacy_impl = resizeOutputForNormal(output, mean, std);
normal_cpu_(output, 0, 1, gen);
if (is_legacy_impl) {
  output.mul_(std.reshape(mean.sizes())).add_(mean);
}
else {
  output.mul_(std).add_(mean);
}
```

Reply:
done